pandas/tests/frame/test_constructors.py

```diff
@@ -1340,6 +1340,26 @@ def test_constructor_list_of_namedtuples(self):
         result = DataFrame(tuples, columns=["y", "z"])
         tm.assert_frame_equal(result, expected)
 
+    def test_constructor_list_of_dict_order(self):
+        # GH10056
+        data = [
+            {"First": 1, "Second": 4, "Third": 7, "Fourth": 10},
+            {"Second": 5, "First": 2, "Fourth": 11, "Third": 8},
+            {"Second": 6, "First": 3, "Fourth": 12, "Third": 9, "YYY": 14, "XXX": 13},
+        ]
+        expected = DataFrame(
+            {
+                "First": [1, 2, 3],
+                "Second": [4, 5, 6],
+                "Third": [7, 8, 9],
+                "Fourth": [10, 11, 12],
+                "YYY": [None, None, 14],
+                "XXX": [None, None, 13],
+            }
+        )
+        result = DataFrame(data)
+        tm.assert_frame_equal(result, expected, check_like=PY36)
```

Comment:
```suggestion
        tm.assert_frame_equal(result, expected, check_like=not PY36)
```

Its on older versions we need to do the check like?

Reply:
we want check_like=true on python>=3.6, right? isn't this right?

Reply:
I think not. We should ensure that both expected and result are correct (preserving the order) for > py36, and then for older version allow different column order. Am I missing something?

Reply:
it is now `not PY36` in all cases, which means (IIUC) that `check_like` is True on Py35, and so column order is ignored on py35. 

Really badly named arg.

Reply:
> Really badly named arg.

pls cut the commentary

Reply:
It's confusing and I'm saying so.

Reply:
Please @pilkibun, you are free to say it (I often see and mention confusing things in pandas as well), but you can also say it in a more respectful way

Reply:
I didn't realize it *was* disrespectful or I wouldn't have said it. I'm still not sure why it triggered jreback. No offense intended. I regularly say similar things about code I've written 6 months ago.